client/landing/stepper/declarative-flow/internals/steps-repository/pattern-assembler/index.tsx

```diff
@@ -367,16 +373,38 @@ const PatternAssembler = ( {
 
 	const onSubmit = () => {
 		const design = getDesign();
+		const stylesheet = design?.recipe?.stylesheet ?? '';
+		const themeId = getThemeIdFromStylesheet( stylesheet );
 
-		if ( ! siteSlugOrId ) {
+		if ( ! siteSlugOrId || ! site?.ID || ! themeId ) {
 			return;
 		}
 
-		setPendingAction( () =>
-			applyThemeWithPatterns( siteSlugOrId, design, syncedGlobalStylesUserConfig ).then(
-				( theme: ActiveTheme ) => reduxDispatch( setActiveTheme( site?.ID || -1, theme ) )
-			)
-		);
+		if ( isEnabled( 'pattern-assembler/logged-in-showcase' ) ) {
+			setPendingAction( () =>
+				Promise.resolve()
+					.then( () =>
+						reduxDispatch(
+							activateOrInstallThenActivate( themeId, site?.ID, 'assembler', false, false )
+						)
+					)
+					.then( () =>
+						assembleSite( siteSlugOrId, isSiteJetpack ? themeId : stylesheet, {
```

Comment:
![badge](https://img.shields.io/badge/review-ask-blue.svg)
What will happen if `/sites/<>/site-assembler` respond with 500? Could the user see a message or something on UI?

Reply:
The [processing-step](https://github.com/Automattic/wp-calypso/blob/a46d327d187c79654b6e8abc18cc8be387cd5657/client/landing/stepper/declarative-flow/internals/steps-repository/processing-step/index.tsx#L76) will handle this kind of error, and we can [decide whether to go when submitting the result](https://github.com/Automattic/wp-calypso/blob/a46d327d187c79654b6e8abc18cc8be387cd5657/client/landing/stepper/declarative-flow/site-setup-flow.ts#L284)

Reply:
That's good to know! Thank you for even telling me where the code is!